File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/preset/InventoryPresetItemUpdateErrorMessage.scala





















// Generated on 05/03/2013 10:34:58
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait InventoryPresetItemUpdateErrorMessage extends MessageDeserializer {
  val messageId = 6211

  protected trait Message extends BaseMessage with Serializable {
    val messageId = InventoryPresetItemUpdateErrorMessage.messageId

    val code: Byte

    def serialize(buf: Buffer) {
      buf.writeByte(code)
    }

  }

  protected def create(code: Byte): Target = throw new IllegalAccessException

  def deserialize(buf: Buffer) = {
    val code = buf.readByte()

    create(code)
  }

}

object InventoryPresetItemUpdateErrorMessage extends InventoryPresetItemUpdateErrorMessage {
  case class MessageImpl(code: Byte) extends Message
  type Target = MessageImpl

  override def create(code: Byte) = MessageImpl(code)

  def apply(code: Byte) = create(code)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/InventoryPresetItemUpdateErrorMessage.java


// Generated on 05/08/2013 19:37:57
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class InventoryPresetItemUpdateErrorMessage extends NetworkMessage {
    public static final int MESSAGE_ID = 6211;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public byte code;
    
    public InventoryPresetItemUpdateErrorMessage() { }
    
    public InventoryPresetItemUpdateErrorMessage(byte code) {
        this.code = code;
    }
    
    @Override
    public void serialize(Buffer buf) {
        buf.writeByte(code);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        code = buf.readByte();
        if (code < 0)
            throw new RuntimeException("Forbidden value on code = " + code + ", it doesn't respect the following condition : code < 0");
    }
    
}
